### PR TITLE
Config path_regex

### PR DESCRIPTION
**Pull request by dstephan1**
- Added path_regex to the config
    - path_regex has priority over path
    - ecmascript syntax
    - path_regex <prefix> <regex> <handler type> {<config block>}
- Refactored HandlerContainer into its own class
- Refactored StaticHandler to be easily inheritable
- Refactored markdown conversion into its own handler MarkdownHandler
    ``` 
    path_regex /static \.md MarkdownHandler {
        root static;
    } 
    ```
Known issues:
- our config parser doesn't have escape sequences, so {} and quotes don't work
- that static file handler gives blank pages for folders now

### CODE REVIEW

**Comment by jteo1** [comment]
Looks good, but what was the reason we moved markdown handler to be separate from the static file handler?

**Comment by dstephan1** [comment]
I separated it to keep the functionality of the static file handler simple. If you wanted to add more specific type handlers the static file handler would quickly become a mess.

File: static_handler.cc [hunk 1]
```diff
@@ -41,76 +40,56 @@ RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix, const
...
-				BOOST_LOG_TRIVIAL(error) << "Unable to open file: " << file_path;
-				NotFoundHandler not_found_handler;
-				return not_found_handler.HandleRequest(request, response);
-			}
+	return RequestHandler::OK;
+}
+
+
+bool StaticHandler::ProcessFile(const std::string& path, const std::string& data, std::string *new_data, std::string *content_type) const{
+	*new_data = data; // there must be a better way
```
Comment (kevinazhu) [inline static_handler.cc]: Why does this need to happen? It doesn't seem like data or new_data are used at all in this function.

Comment (dstephan1) [inline static_handler.cc]: They aren't. This really only exists so that things that inherit the static file handler can override the behavior.

Comment (kevinazhu) [inline static_handler.cc]: Ohhh that makes sense

**Pull request merged**